cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/RDDCollection.java

```diff
@@ -176,9 +196,28 @@ public RDDCollection(JavaSparkExecutionContext sec, FunctionCache.Factory functi
 
       Iterator<Column> leftIter = leftJoinColumns.iterator();
       Iterator<Column> rightIter = rightJoinColumns.iterator();
-      Column joinOn = eq(leftIter.next(), rightIter.next(), joinRequest.isNullSafe());
-      while (leftIter.hasNext()) {
-        joinOn = joinOn.and(eq(leftIter.next(), rightIter.next(), joinRequest.isNullSafe()));
+
+      Column joinOn;
+      if (atLeastOneTableRequired) {
```

Comment:
You need to update this on the go (I believe you can use atLeastOneTableRequired), otherwise 
A FJ B FJ C RJ D will have problems on C.
Also please add tests.

Reply:
fixed and added in test case as well.